### ISSUE TITLE
PassportConfigurator domain option conflict

### ISSUE THREAD

**Issue by clockworkgr** [strongloop/loopback-component-passport#129]
I might be totally minsuderstanding this but I believe there is an issue with the domain option in providers.json.

According to passport-configurator.js in the default auth callback, if the domain option is set, it is used to set the domain of the cookie.

However, certain passport strategies (passport-ntlm, passport-auth0 and possibly others) use the domain option to set the server against which auth will be made.

In my case, I am trying to authenticate using auth0.com's passport-auth0 which requires a domain option like AUTH0_ACCOUNT_NAME.REGION.auth0.com which leads to the user/accessToken cookie to be set to that domain by the PassportConfigurator default callback.

If I'm looking at this the right way, and the  domain option in loopback-component-passport is a loopback specific thing, the option should probably be prefixed somehow to avoid such conflicts (e.g. lb_cookie_domain)

Can someone confirm the above is correct (and therefore an issue that needs to be resolved) ?

**Comment by jeff3yan** [comment]
I have confirmed the exact issue and reproduced it using the auth0 login strategy. Other strategies such as facebook or twitter don't use the domain option in their providers.json configuration, avoiding this issue.

**Comment by jannyHou** [comment]
@jeff3yan @clockworkgr could you upload the code to reproduce this error by forking [sandbox](https://github.com/strongloop/loopback-sandbox) and sending a pr?
Or explain how to reproduce it if the code is not available.

Thanks.

**Comment by jeff3yan** [comment]
There's not much code to reproduce the issue. I cloned the [loopback-example-passport](https://github.com/strongloop/loopback-example-passport) repository, then add a /providers.json file to use an auth0 passport strategy.

1. Clone the repository and `npm install`
2. `npm install --save passport-auth0`
3. Create a providers.json with the following, and enter in an auth0 access ID and secret.
`
{
  "auth0-login": {
    "provider": "auth0",
    "module": "passport-auth0",
    "clientID": "AUTH0_ACCESS_ID",
    "clientSecret": "AUTH0_SECRET",
    "domain": "jeff3yan.auth0.com",
    "callbackURL": "/auth/auth0/callback",
    "authPath": "/auth/auth0",
    "callbackPath": "/auth/auth0/callback",
    "successRedirect": "/auth/account",
    "failureRedirect": "/login",
    "scope": ["email"],
    "failureFlash": true
  }
}
`
4. Inside /server/views/pages/login.jade, add/replace one of the existing login buttons:
`a.btn.btn-primary(href="/auth/auth0") Login with Auth0`
5. Start the app with `node .`
6. Browse to http://localhost:3000, then try login using the new auth0 button
7. Notice that the access_key and userId cookies have not been set (which are set if a Facebook strategy is used)

The exact part of code seems to have been added on #75, where options.domain is used to set the cookie domain.
